app/models/contract.rb

```diff
@@ -35,7 +35,7 @@ class Contract < ApplicationRecord
   validates :user_key, length: { maximum: 256 }
 
   # TODO: rename to buyer_account and remove alias
-  belongs_to :user_account, class_name: 'Account', autosave: false
+  belongs_to :user_account, class_name: 'Account', autosave: false, inverse_of: :contracts
```

Comment:
I noticed that Rails didn't figure out the reverse association here, so helping it do.